src/librustc_front/lowering.rs

```diff
@@ -140,6 +141,11 @@ impl<'a, 'hir> LoweringContext<'a> {
             result
         }
     }
+
+    // panics if this LoweringContext's NodeIdAssigner is not a Session
```

Comment:
I think the bit about being a Session is too much detail, one could imagine implementing NodeIdAssigner with something that is not a Session but can still provide a diagnostic (also, nit, comments should start with a capital and end with a period).